### PR TITLE
check bannerImage & welcomeBannerImage

### PR DESCRIPTION
**Pull request by amadotejada**
The Dialog fails to open if the banner/welcome images are not available. We use this to test the image URLs and set an alternative image if the default fails. Folks can add a second custom image as a backup.

### CODE REVIEW

**Comment by dan-snelson** [comment]
Thanks, @amadotejada. 

https://github.com/bartreardon/swiftDialog/issues/239

File: Setup-Your-Mac-via-Dialog.bash [hunk 1]
```diff
@@ -397,6 +397,14 @@ welcomeBannerText="Happy $( date +'%A' ), ${loggedInUserFirstname} and welcome t
 welcomeCaption="Please review the above video, then click Continue."
 welcomeVideoID="vimeoid=812753953"
 
+# Check if the custom welcomeBannerImage is available, and if not, use a alternative image
+if curl --output /dev/null --silent --head --fail "$welcomeBannerImage"; then
+    echo "welcomeBannerImage is available"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:402]: Please replace `echo` with `updateScriptLog "WELCOME DIALOG: …"

File: Setup-Your-Mac-via-Dialog.bash [hunk 2]
```diff
@@ -397,6 +397,14 @@ welcomeBannerText="Happy $( date +'%A' ), ${loggedInUserFirstname} and welcome t
...
+else
+    echo "welcomeBannerImage is not available, using alternative image"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:404]: Please replace `echo` with `updateScriptLog "WELCOME DIALOG: …"

File: Setup-Your-Mac-via-Dialog.bash [hunk 3]
```diff
@@ -524,6 +532,14 @@ bannerText="Setting up ${loggedInUserFirstname}'s Mac"
 helpmessage="If you need assistance, please contact the Global Service Department:  \n- **Telephone:** +1 (801) 555-1212  \n- **Email:** support@domain.org  \n- **Knowledge Base Article:** KB0057050  \n\n**Computer Information:**  \n- **Operating System:**  ${macOSproductVersion} (${macOSbuildVersion})  \n- **Serial Number:** ${serialNumber}  \n- **Dialog:** ${dialogVersion}  \n- **Started:** ${timestamp}"
 infobox="Analyzing input …" # Customize at "Update Setup Your Mac's infobox"
 
+# Check if the custom bannerImage is available, and if not, use a alternative image
+if curl --output /dev/null --silent --head --fail "$bannerImage"; then
+    echo "bannerImage is available"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:537]: Please replace `echo` with `updateScriptLog "WELCOME DIALOG: …"

File: Setup-Your-Mac-via-Dialog.bash [hunk 4]
```diff
@@ -524,6 +532,14 @@ bannerText="Setting up ${loggedInUserFirstname}'s Mac"
...
+else
+    echo "bannerImage is not available, using alternative image"
```
Comment (dan-snelson) [inline Setup-Your-Mac-via-Dialog.bash:539]: Please replace `echo` with `updateScriptLog "WELCOME DIALOG: …"

**Review by dan-snelson — COMMENTED** [review]
@amadotejada:

The `development` branch is now at `1.10.0-rc4`; will you please update your side to this version and re-submit?

Thanks.


--
Dan

**Comment by amadotejada** [comment]
@dan-snelson

Updated to 1.10.0-rc4 and replaced the `echo` lines. 

Thank you

**Review by dan-snelson — CHANGES_REQUESTED** [review]
@amadotejada:

Perhaps it's me, but it _appears_ that this is still committed against `1.9.0`:

https://github.com/amadotejada/Setup-Your-Mac/blob/647241c9c263adf7ef7eca582390137bf7a25b74/Setup-Your-Mac-via-Dialog.bash#L36

**Comment by dan-snelson** [comment]
Closing; see [Pull No. 22](https://github.com/dan-snelson/Setup-Your-Mac/pull/22).

**Pull request closed without merging**